src/Analyzers/Core/Analyzers/ForEachCast/AbstractForEachCastDiagnosticAnalyzer.cs

```diff
@@ -31,6 +31,7 @@ protected AbstractForEachCastDiagnosticAnalyzer()
                   diagnosticId: IDEDiagnosticIds.ForEachCastDiagnosticId,
                   EnforceOnBuildValues.ForEachCast,
                   CodeStyleOptions2.ForEachExplicitCastInSource,
+                  language: null!,
```

Comment:
So this is the part that I dislike - the analyzer supports C# and VB and we are passing in null. It makes me want to convert the new property being added to an array, with empty array indicating all languages.

Reply:
Ah, makes sense then! Follow-up PR is fine.

Reply:
Actually, another question. What if we wanted to add a new `ISingleValuedOption` that makes sense for C# and VB, has the same value for both languages, but is unsupported for other languages like F#, TS, etc.? Encoding a single LanguageName instead of an array of LanguageNames would mean that the option client needs to deduce what _all languages_ represents for such an option (which might be acceptable). I don't know if this scenario arises though, so may be it is fine to go single LanguageName route for now.

Reply:
I don't know enough about other languages' interaction with Roslyn. But the current uses of `LanguageName` property are for IDExxxx and editorconfig UI, these are C# and/or VB only.

So it should be good for now, and if in future this turned to be not sufficient, the design can be revised.